src/lib/dd_profiling.cc

```diff
@@ -248,6 +249,13 @@ static int ddprof_start_profiling_internal() {
         flags |= ddprof::AllocationTracker::kDeterministicSampling;
```

Comment:
@nsavoire what should the deterministic option be like ? Should there be a matching sampling option flags in the input ? In which case I should rebase on @sanchda 's PR